skimage/metrics/overlap.py

```diff
@@ -0,0 +1,221 @@
+"""
+Module defining a number of functions to quantify the overlap between shapes.
+for instance rectangles representing detections by bounding-boxes.
+
+"""
+
+class Rectangle():
+    """
+    Represents rectangles which position corresponds to the (r,c) coordinates for the top left corner.
+    Rectangle objects have a number of attributes:
+        - coordinates for the bottom right corner
+        - rectangle dimensions
+        - area
+    as well as utility functions :
+        - computation of the intersection of 2 rectangles
+        - equivalent area for te union of 2 rectangles.
```

Comment:
This should use NumPy style formatting, see https://numpy.org/doc/stable/docs/howto_document.html

and a good example:

https://github.com/scikit-learn/scikit-learn/blob/7e197fd600b72483ff2343b8d2605368d1ab24bf/sklearn/svm/_classes.py#L13

Reply:
see below 😉